test/chainedvector.jl

```diff
@@ -423,6 +436,110 @@
+            [24, 28, 75, 42, 7, 38, 59, 10, 21, 30, 14, 8, 39, 13, 68, 42],
+        ChainedVector([[2.1, -4.6, -2.5], [-5.0, 6.4, 2.0, -0.5], [-6.1, -7.6, -3.2, -4.7, 4.3], [-1.7, 6.4, -8.9, -7.4], [-7.7, -1.4, 3.1, 4.5]]) =>
+            [2.1, -4.6, -2.5, -5.0, 6.4, 2.0, -0.5, -6.1, -7.6, -3.2, -4.7, 4.3, -1.7, 6.4, -8.9, -7.4, -7.7, -1.4, 3.1, 4.5],
+        ChainedVector([[-8.5, -1.2, -3.8, 7.5], [8.2, 7.5, -5.3], [-2.7, 0.6, -6.2, 6.1, 1.4]]) =>
+            [-8.5, -1.2, -3.8, 7.5, 8.2, 7.5, -5.3, -2.7, 0.6, -6.2, 6.1, 1.4],
+        ChainedVector([[-7.2], [8.1, 2.3, 7.5], [-8.4, -5.7]]) =>
+            [-7.2, 8.1, 2.3, 7.5, -8.4, -5.7],
+        ChainedVector([[-3.7, 7.8, -5.0], [0.1], [5.0, -4.1], [-1.6, -0.9, 8.7, -7.8]]) =>
+            [-3.7, 7.8, -5.0, 0.1, 5.0, -4.1, -1.6, -0.9, 8.7, -7.8],
+        ChainedVector([[8.6, -2.0], [8.0, 3.4, 3.3], [1.0], [5.4, -2.6, -4.7, 4.4, 4.4], [7.9]]) =>
+            [8.6, -2.0, 8.0, 3.4, 3.3, 1.0, 5.4, -2.6, -4.7, 4.4, 4.4, 7.9],
+        ChainedVector([[7.6, 5.9], [7.9, -8.8, -1.5, -0.4, 6.0], [-5.1, -0.4, 4.4, 7.3]]) =>
+            [7.6, 5.9, 7.9, -8.8, -1.5, -0.4, 6.0, -5.1, -0.4, 4.4, 7.3],
+        ChainedVector([[3.2, -3.2, 1.2, -1.2, -2.1], [0.5], [6.2], [2.9], [-8.1, 5.8, 4.8, -3.4, -3.1]]) =>
+            [3.2, -3.2, 1.2, -1.2, -2.1, 0.5, 6.2, 2.9, -8.1, 5.8, 4.8, -3.4, -3.1],
+        ChainedVector([[-8.0, -1.9, -5.1, -1.4, -8.3], [5.1, -3.7, 6.3, -4.8, -3.3], [-7.0], [-2.4, 4.0, -3.7], [-6.6, -6.9, 2.5, -1.3]]) =>
+            [-8.0, -1.9, -5.1, -1.4, -8.3, 5.1, -3.7, 6.3, -4.8, -3.3, -7.0, -2.4, 4.0, -3.7, -6.6, -6.9, 2.5, -1.3],
+        ChainedVector([[-7.5], [-1.5, -5.8, 8.4], [-8.4, -1.9, 2.3, -0.8, -8.5], [0.2, 0.5, -7.4, 2.1, -3.9]]) =>
+            [-7.5, -1.5, -5.8, 8.4, -8.4, -1.9, 2.3, -0.8, -8.5, 0.2, 0.5, -7.4, 2.1, -3.9],
+        ChainedVector([[3.9, -8.9], [-0.3, 0.0, 7.3], [-2.9, 8.6, 5.8, 0.5], [0.0, -4.5, 3.3, 0.4, -3.2]]) =>
+            [3.9, -8.9, -0.3, 0.0, 7.3, -2.9, 8.6, 5.8, 0.5, 0.0, -4.5, 3.3, 0.4, -3.2],
+    ]
+    for (x,y) in test_vectors
+        try
+            @test length(x) == length(y)
+            # should this be approx?
+            @test sum(x) ≈ sum(y)
+            @test findmax(x) == findmax(y)
+            @test findmin(x) == findmin(y)
+            @test maximum(x) == maximum(y)
+            @test minimum(x) == minimum(y)
+            @test argmax(x) == argmax(y)
+            @test argmin(x) == argmin(y)
+            @test findmax(x->x+1, x) == findmax(x->x+1, y)
+            @test findmin(x->x-1, x) == findmin(x->x-1, y)
+            @test findfirst(isodd, x) == findfirst(isodd, y)
+            @test findfirst(iseven, x) == findfirst(iseven ,y)
+            @test findlast(isodd, x) == findlast(isodd, y)
+            @test findlast(iseven, x) == findlast(iseven ,y)
+            @test findnext(isodd, x, 5) == findnext(isodd, y, 5)
+        catch err
+            error("Test failed for $x")
+        end
+    end
```

Comment:
```suggestion
    @testset for vecs in test_vectors
        x, y = vecs
        @test length(x) == length(y)
        # should this be approx?
        @test sum(x) ≈ sum(y)
        @test findmax(x) == findmax(y)
        @test findmin(x) == findmin(y)
        @test maximum(x) == maximum(y)
        @test minimum(x) == minimum(y)
        @test argmax(x) == argmax(y)
        @test argmin(x) == argmin(y)
        @test findmax(x->x+1, x) == findmax(x->x+1, y)
        @test findmin(x->x-1, x) == findmin(x->x-1, y)
        @test findfirst(isodd, x) == findfirst(isodd, y)
        @test findfirst(iseven, x) == findfirst(iseven ,y)
        @test findlast(isodd, x) == findlast(isodd, y)
        @test findlast(iseven, x) == findlast(iseven ,y)
        @test findnext(isodd, x, 5) == findnext(isodd, y, 5)
    end
```

This syntax should be supported already by Julia 1.3; might be good to replace the big double setup with just the `ChainedVector`s and `collect` them explicitly in the loop here.

Reply:
The main reason I'm not using collect is because I don't trust it to be correct here.

Reply:
Resolved in 
b5a923f

Reply:
I think we can assume it to be correct

```
julia> using Supposition

julia> using SentinelArrays

julia> data_vector = Data.Vectors(Data.Integers{UInt8}(); max_size=100);

julia> vecs = Data.Vectors(data_vector; min_size=1, max_size=10);

julia> chainedvecs = map(vecs) do vs
           all(isempty, vs) && reject!()
           ChainedVector(vs)
       end;

julia> @check max_examples=100_000 collect_consistent(cv=chainedvecs) -> all(splat(==), zip(cv, collect(cv)));
Test Summary:      | Pass  Total  Time
collect_consistent |    1      1  0.9s
```